### ISSUE TITLE
Website --web.enable-admin-api confusion

### ISSUE THREAD

**Issue by ericdstein** [thanos-io/thanos#3278]
**Thanos, Prometheus and Golang version used**:
Thanos: v0.15
Prometheus: 2.21

In multiple places, including the [sidecar docs](https://github.com/thanos-io/thanos/blob/master/docs/components/sidecar.md), I have seen the following statement. 
> The --web.enable-admin-api flag is enabled to support sidecar to get metadata from Prometheus like external labels.

It doesn't appear, however, that this flag controls access to metadata. Instead, it appears to [control access to perform certain admin tasks](https://github.com/prometheus/prometheus/blob/8dbfa14607a5cb138df43638f06c1a938c73df84/web/api/v1/api.go#L315) and thus may only be required if using the Sidecar to send data to object storage. If one were to just use Query, is this flag required to be set for Prometheus?

**Comment by yeya24** [comment]
Right, it is confusing and I believe this flag is not needed. Prometheus external labels are fetched from `/api/v1/status/config` directly.

**Comment by bwplotka** [comment]
Not really, we use flags endpoint as well.

**Comment by bwplotka** [comment]
But overall, **let's double check** we added this piece of doc and requirement at the very start of Thanos, I suppose for snapshot purposes, although we don't use it now on sidecar. So there is some chances that this flag is not needed at all. (:

**Comment by ericdstein** [comment]
Thank you both! :-) I found the reason the statement was added. Prior to https://github.com/thanos-io/thanos/issues/1354 https://github.com/thanos-io/thanos/pull/1355, the ExternalLabels func returned error `is 'web.enable-admin-api' flag enabled? got non-200 response code`. The recommendation was added when someone encountered this error.

The only func I could find that calls an admin api is the Snapshot func in promclient. But, as was stated, I couldn't find where that func is currently used.
